src/main/java/io/kestra/plugin/gcp/firestore/Query.java

```diff
@@ -0,0 +1,230 @@
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static io.kestra.core.utils.Rethrow.throwConsumer;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+        title = "Query documents of a collection.",
+        description = "Query documents of a collection."
+)
+@Plugin(
+        examples = {
+                @Example(
+                        title = "Query documents of a collection.",
+                        code = {
+                                "collection: \"persons\"",
+                                "field: \"firstname\"",
+                                "value: \"Doe\""
+                        }
+                )
+        }
+)
+public class Query extends AbstractFirestore implements RunnableTask<Query.Output> {
+
+    @Schema(
+            title = "Whether to store the data from the query result into an ion serialized data file."
+    )
+    @PluginProperty(dynamic = false)
+    @Builder.Default
+    private Boolean store = false;
+
+    @Schema(
+            title = "Field name for the where clause."
+    )
+    @PluginProperty(dynamic = false)
+    @NotNull
+    private String field;
+
+    @Schema(
+            title = "Field value for the where clause.",
+            description = "Field value for the where clause. Only strings are supported at the moment."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    private String value;
+
+    @Schema(
+            title = "The query operator for the where clause, by default EQUAL_TO that will call 'collection.whereEqualTo(name, value)'",
+            description = "Can be one of EQUAL_TO, " +
+                    "NOT_EQUAL_TO, " +
+                    "LESS_THAN, " +
+                    "LESS_THAN_OR_EQUAL_TO, " +
+                    "GREATER_THAN, " +
+                    "GREATER_THAN_OR_EQUAL_TO"
+    )
+    @PluginProperty(dynamic = false)
+    @Builder.Default
+    private QueryOperator queryOperator = QueryOperator.EQUAL_TO;
+
+    @Schema(
+            title = "Field name for the order by clause."
+    )
+    @PluginProperty(dynamic = false)
+    private String orderBy;
+
+    @Schema(
+            title = "Start offset for pagination of the query results."
+    )
+    @PluginProperty(dynamic = true)
+    private Integer offset;
+
+    @Schema(
+            title = "Maximum numbers of returned results."
+    )
+    @PluginProperty(dynamic = true)
+    private Integer limit;
+
+    @Override
+    public Query.Output run(RunContext runContext) throws Exception {
+        try(var firestore = this.connection(runContext)) {
+            var collectionRef = this.collection(runContext, firestore);
+
+            var query = getQuery(runContext, collectionRef, this.queryOperator);
+            if(this.orderBy != null) {
+                query.orderBy(this.orderBy);
+            }
```

Comment:
I'm not a  big fan of those, can we use Optional on the plugin properties or not ? I didn't find any example of how to deal with optional params

Reply:
OK